bintool/bintool.cpp

```diff
@@ -421,10 +444,32 @@ block place_new_block(std::vector<uint8_t> &bin, uint32_t storage_addr, std::uni
 
     // loop back to first block
     block new_block(new_block_addr, loop_start_rel);
-    // copt the existing block
-    std::copy(first_block->items.begin(),
-              first_block->items.end(),
-              std::back_inserter(new_block.items));
+    // check if last block has an image_def
```

Comment:
From a quick glance, there seems to be a lot of overlap in the change on lines 317-342 and the change on lines 447-472. Does that suggest that a slight refactoring might be appropriate?

Reply:
There are quite a few functions in bintool which are duplicated for ELFs and BINs, which share some code but with key differences. Some of it could probably be refactored to remove the duplication, but I think that's a change for a different PR, to do that refactoring for all the functions in bintool rather than just the ones changed in this PR.

Reply:
#231